software/ADPL_electron/ADPL_electron.ino

```diff
@@ -200,9 +213,14 @@ int publish_data(int last_publish_time) {
             tempHXCI.temp, tempHXCO.temp, tempHTR.temp, tempHXHI.temp, tempHXHO.temp,
             int(valve.gasOn), int(bucket.tip_count));
 
-    publish_success = Particle.publish("DATA",data_str);
+    if (sdFile) {
+        sdFile.println(data_str);
```

Comment:
is there an API for us to "flush the buffer" and force a blocking write to the filesystem here so we don't lose data?